packages/twenty-front/src/pages/settings/data-model/SettingsObjectTable.tsx

```diff
@@ -249,10 +251,16 @@ export const SettingsObjectTable = ({
               }
               link={
                 isActive
-                  ? getSettingsPath(SettingsPath.ObjectDetail, {
-                      objectNamePlural:
-                        objectSettingsItem.objectMetadataItem.namePlural,
-                    })
+                  ? isDefined(applicationId)
+                    ? getSettingsPath(SettingsPath.ApplicationObjectDetail, {
```

Comment:
I don't think we should do a different page for Application and non-Application objects, it seems too similar. Will keep reading the PR but this seems odd on first read, maybe I'll change my mind